docs/entityoperator.md

```diff
@@ -0,0 +1,234 @@
+| Action    | Operator Responsible |
+| -------- | ------- |
+| Developer creates a `KafkaTopic` CR  | Topic Operator creates & syncs the topic in Kafka    |
+| Developer creates a KafkaUser CR | User Operator creates the user & credentials  |
+| Application retrieves credentials from Kubernetes Secrets	| Application mounts the secrets for authentication |
+| Application connects to Kafka using these credentials | Producer/Consumer communicates with Kafka  |
+
+
+## Impact of Central Cluster Failure on Kafka Clients in a Stretch Cluster
+
+In stretch Kafka deployment, where
+
+- ✅ Kafka brokers and controllers are spread across multiple Kubernetes clusters.
+- ✅ The central cluster hosts all Kafka CRs, including Kafka, KafkaNodePool, KafkaUser, and KafkaTopic.
+- ✅ The Entity Operator (managing users & topics) runs in the central cluster.
+
+## Does the Kafka Cluster Still Function?
+
+Yes, but with conditions.
+
+- Brokers in other Kubernetes clusters are still running.
+- Kafka controllers may still be running (if a quorum is maintained).
+
+Replication between brokers (spread across clusters) continues as long as there’s a majority quorum of controllers.
+
+- ✅ If a majority of controllers are still available in the surviving clusters, Kafka continues running.
+- 🚨 If the majority of controllers were in the central cluster and lost, Kafka will experience a complete outage.
+
+💡 Mitigation -> user need to ensure the `controller.quorum.voters` are spread across clusters to avoid losing the majority.
+
+## Can Kafka Clients Still Communicate with Brokers?
+
+Yes, if at least one broker remains reachable.
+
+Clients connect to brokers, not the Entity Operator or the central cluster.
+
+- If clients' bootstrap servers list includes brokers running in the surviving clusters, they can still connect.
+- If a leader partition for a topic is hosted on a broker in a surviving cluster, the topic remains available.
+- If all leader partitions for a topic were on brokers in the central cluster, that topic becomes unavailable.
+
+💡 Mitigation -> Use Kafka’s rack awareness (`broker.rack`) and ISR (In-Sync Replica) balancing to distribute leader partitions across clusters.
+
+## Can Clients Still Produce & Consume Messages?
+
+Yes, if leader partitions are available.
+No, if all leader partitions were in the lost central cluster.
+
+**Producing Messages**
+
+- A producer can send messages only if the leader partition for a topic is still available in a surviving cluster.
+- If the leader was in the lost central cluster, Kafka automatically elects a new leader (if ISR exists).
+- If no ISR exists, production fails.
+
+**Consuming Messages**
+
+- Consumers can still fetch messages from available partitions.
+- If consumer group metadata was stored in a lost broker, the group might experience issues.
+
+💡 Mitigation -> Enable `min.insync.replicas` and leader election across clusters to ensure partition availability.
+
+
+## What About Entity Operator Functions?
```

Comment:
IMO this is the important section in this page as the rest is documenting our understanding of how EO works and generic stretch cluster aspects.